python/grep.py

```diff
@@ -198,7 +206,7 @@
 ###
 
 from os import path
-import sys, getopt, time, os, re, tempfile
+import sys, getopt, time, os, re, cPickle
```

Comment:
cPickle seems to be a python 2 thing. In python 3 it's called pickle.

Perhaps you should do something like 

```py
pycompat = sys.version_info[0] < 3
if pycompat:
  import cPickle as pickle
else:
  import pickle

# and adapt the code to work in both py2 and py3 where API changes are necessary.
```

Reply:
Without this compatibility it is a breaking change as I think grep works on python 3. If it doesn't, movement towards 3.x support is still something good to be working towards.

Reply:
I added a fallback to `pickle` in case importing `cPickle` fails